deployment/terraform/output.go

```diff
@@ -126,13 +126,17 @@ func (t *Terraform) loadOutput() error {
 	return nil
 }
 
-// Output reads the current terraform output
-func (t *Terraform) Output() (*Output, error) {
-	var err error
+func (t *Terraform) setOutput() error {
 	if t.output == nil {
-		err = t.loadOutput()
+		return t.loadOutput()
 	}
-	return t.output, err
+	return nil
+}
+
+// Output reads the current terraform output and caches it internally for future use.
+// The output is guaranteed to be up to date after calls to Create and Destroy.
+func (t *Terraform) Output() (*Output, error) {
+	return t.output, t.setOutput()
```

Comment:
This seems to be working although I am not sure if it's defined behaviour. Please let me know if it's better/safer to store the result in a variable first.

Reply:
Functionally it's the same. But it does look a bit odd to read at first.

Something like this has more repetition, but looks more readable. 0/5, up to you

```
var err error
if t.output == nil {
  err = t.loadOutput()
}
return t.output, err
```